prov/gni/src/gnix_ep.c

```diff
@@ -2112,16 +2110,21 @@ DIRECT_FN int gnix_ep_open(struct fid_domain *domain, struct fi_info *info,
 	ep_priv->requires_lock = (domain_priv->thread_model !=
 				  FI_THREAD_COMPLETION);
 	ep_priv->info = fi_dupinfo(info);
+	ep_priv->info->addr_format = info->addr_format;
+
+	GNIX_DEBUG(FI_LOG_DEBUG, "ep(%p) is using addr_format(%s)\n", ep_priv,
+		  ep_priv->info->addr_format == FI_ADDR_STR ? "FI_ADDR_STR" :
+		  "FI_ADDR_GNI");
 
-	if (ep_priv->info->src_addr)
-		memcpy(&ep_priv->src_addr,
-		       ep_priv->info->src_addr,
+	if (info->src_addr) {
+		memcpy(&ep_priv->src_addr, info->src_addr,
 		       sizeof(struct gnix_ep_name));
+	}
 
-	if (ep_priv->info->dest_addr)
-		memcpy(&ep_priv->dest_addr,
-		       ep_priv->info->dest_addr,
+	if (info->dest_addr) {
+		memcpy(&ep_priv->dest_addr, info->dest_addr,
```

Comment:
You are doing quite a bit of cleanup work here. It would be best if cleanup was separate from feature work so that it was easily identifiable by git blame, what the change was related to.